src/HackerNews/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,192 @@
+
+  const [news, setNews] = useState<Array<response>>(() => {
+    return [];
+  });
+  const [newsIds, setNewsIds] = useState(() => {
+    return [];
+  });
+  const [selectedNew, setSelectedNew] = useState<response>(() => {
+    return {
+      by: "",
+      id: -1,
+      descendants: -1,
+      kids: [],
+      time: -1,
+      title: "",
+      type: "",
+      url: "",
+      score: -1,
+    };
+  });
+
+  useEffect(() => {
+    fetch(url)
+      .then((response) => response.json())
+      // Костыль: у веб апи нельзя определить кол-во постов, получаем 500, выкидываю 400
+      .then((json) => {
+        setNewsIds(() => {
+          return json.slice(0, 100);
+        });
+      });
+
+    setInterval(() => {
+      if (window.location.pathname === "/") {
+        setNewsIds(() => {
+          return [];
+        });
+        setNews(() => {
+          return [];
+        });
+        fetch(url)
+          .then((response) => response.json())
+          // Костыль: у веб апи нельзя определить кол-во постов, получаем 500, выкидываю 400
+          .then((json) => {
+            setNewsIds(() => {
+              return json.slice(0, 100);
+            });
+          });
+      }
+    }, 60000);
+  }, []);
+
+  useEffect(() => {
+    newsIds.forEach((id) => {
+      let detailsUrl = `https://hacker-news.firebaseio.com/v0/item/${id}.json?print=pretty`;
+      fetch(detailsUrl)
+        .then((response) => response.json())
+        .then((json) => {
+          setNews((prevState) => {
+            return [...prevState, json];
+          });
+        });
+    });
+  }, [newsIds]);
+
+  const refreshButtonClick = (): void => {
+    setNewsIds(() => {
+      return [];
+    });
+    setNews(() => {
+      return [];
+    });
+
+    fetch(url)
+      .then((response) => response.json())
+      // Костыль: у веб апи нельзя определить кол-во постов, получаем 500, выкидываю 400
+      .then((json) => {
+        setNewsIds(() => {
+          return json.slice(0, 100);
+        });
+      });
+  };
+
+  const setSelectedNewCallback = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    obj: response
+  ): void => {
+    setSelectedNew(() => {
+      return obj;
+    });
+  };
+
+  if (newsIds.length > 0) {
+    return (
+      <Routes>
+        <Route
+          path="/"
+          element={
+            <DivWrapper>
+              <NewsListHeader>
+                <HeaderText>Welcome to Hacker news (new!)</HeaderText>
+              </NewsListHeader>
+
+              <ButtonContainer>
+                <RefreshButton onClick={refreshButtonClick}>
+                  <ButtonText>Refresh!</ButtonText>
+                </RefreshButton>
+              </ButtonContainer>
+
+              {news.map((elem: response, index) => {
+                return (
+                  <NewsSummary
+                    data={elem}
+                    number={index}
+                    select={setSelectedNewCallback}
+                  ></NewsSummary>
```

Comment:
Юзай самозакрывающие теги если children не нужен
```suggestion
                  />
```